test/reporters/list.spec.js

```diff
@@ -138,10 +138,15 @@ describe('List reporter', function() {
       var actual = {a: 'actual'};
       var expected = {a: 'expected'};
       var checked = false;
+      var stringifyDiffs = 'stringifyDiffObjs';
       var err;
       test = {};
       runner.on = runner.once = function(event, callback) {
-        if (!checked && event === 'fail') {
+        if (
+          callback.toString().includes(stringifyDiffs) && // handle correct callback
```

Comment:
Sorry. Still hate this. Are we sure this is the **only** reporter affected? No other way to make this work correctly?